drivers/adc/adc_ltc2451.c

```diff
@@ -0,0 +1,107 @@
+/* LLTC LTC2451 ADC
+ *
+ * Copyright (c) 2023 Brill Power Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ltc2451, CONFIG_ADC_LOG_LEVEL);
+
+#define DT_DRV_COMPAT lltc_ltc2451
+
+struct ltc2451_config {
+	struct i2c_dt_spec i2c;
+	uint8_t conversion_speed;
+};
+
+static int ltc2451_channel_setup(const struct device *dev,
+				const struct adc_channel_cfg *channel_cfg)
+{
+	(void)dev;
+	(void)channel_cfg;
+	return 0; /* there is only 1 channel which requires no real setup */
+}
+
+static int ltc2451_set_conversion_speed(const struct device *dev, uint8_t conversion_speed)
+{
+	const struct ltc2451_config *config = dev->config;
+	uint8_t speed_bit = -1;
+
+	if (conversion_speed == 60) {
+		speed_bit = 0;
+	} else if (conversion_speed == 30) {
+		speed_bit = 1;
+	} else {
+		err = -EINVAL;
+		LOG_ERR("Invalid conversion speed selected");
+		return err;
+	}
+
+	uint8_t wr_buf[1] = {speed_bit};
+
+	int err = i2c_write_dt(&config->i2c, wr_buf, sizeof(wr_buf));
+
+	if (err == 0) {
+		config->conversion_speed = conversion_speed;
+	} else {
+		LOG_ERR("LTC write failed (err %d)", err);
+	}
+
+	return err;
```

Comment:
`-1` should not be assigned to a `uint8_t`. And why not more simple? 

```suggestion
	const struct ltc2451_config *config = dev->config;
	uint8_t wr_buf[1];
	int err;

	if (conversion_speed == 60) {
		wr_buf[0] = 0;
	} else if (conversion_speed == 30) {
		wr_buf[0] = 1;
	} else {
		LOG_ERR("Invalid conversion speed selected");
		return -EINVAL;
	}

	err = i2c_write_dt(&config->i2c, wr_buf, sizeof(wr_buf));
	if (err != 0) {
		LOG_ERR("LTC write failed (err %d)", err);
	}

	return err;
```